coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -3,50 +3,44 @@
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
+      $http.get(projectDir + $scope.Constants.file_data)
+        .success(function(file_data) {
+            $scope.file_dict = {};
+            Object.keys(file_data).forEach(function(file_name) {
+              if (!(file_name in $scope.file_dict)) {
+                $scope.file_dict[file_name]=file_data[file_name];
+              }
+            });
+          coalaJSON = coalaJSON || projectDir + $scope.Constants.coala;
+          $scope.data = [];
+          $http.get(coalaJSON).success(function(coala_json) {
+            var knownFiles = {};
+            var parseResult = function(el) {
+                el.affected_code.forEach(function(sourceRange){
+                    if (!(sourceRange.file in knownFiles)){
+                        knownFiles[sourceRange.file] = [];
+                    }
+                    knownFiles[sourceRange.file].push({
+                      "start":    sourceRange.start.line,
+                      "end":      sourceRange.end.line,
+                      "diffs":    el.diffs,
+                      "message":  el.message,
+                      "origin":   el.origin,
+                      "severity": el.severity
                     });
+                });
+            };
+            for (var section in coala_json.results) {
+              if (coala_json.results[section].length){ // If error exists
+                coala_json.results[section].forEach(parseResult);
               }
-              knownFiles[sourceRange.file].push({
-                "start":    sourceRange.start.line,
-                "end":      sourceRange.end.line,
-                "diffs":    el.diffs,
-                "message":  el.message,
-                "origin":   el.origin,
-                "severity": el.severity
-              });
+            }
+            $scope.data.push(knownFiles);
           });
-        };
-
-        for (var errors in response.results) {
-          if (response.results[errors][0]){ // If error exists
-              response.results[errors].forEach(parseResult);
-          }
-        }
-
-        $scope.data.push(knownFiles);
-      });
-    };
-    parseCoalaProject("tests/test_projects/simple");
-
-    $scope.range = function(min, max, step) {
-      step = step || 1;
-      var input = [];
-      for (var i = min; i <= max; i += step) {
-        input.push(i);
-      }
-      return input;
-    };
+        });
+      };
+    $http.get("data/Constants.json").success(function(constants) {
+      $scope.Constants = constants;
+      parseCoalaProject($scope.Constants.data);
```

Comment:
Are you sure this code duplication cannot be removed. Maybe, with angular service?